### PR TITLE
216756 change order (downsizing).

### PR DESCRIPTION
**Pull request by ffaraone**

### CODE REVIEW

File: adobe_vipm/adobe/errors.py [hunk 1]
```diff
@@ -1,19 +1,48 @@
...
         if self.details:
             message = f"{message}: {', '.join(self.details)}"
         return message
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         return str(self.payload)
+
+
+def wrap_http_error(func: Callable[..., Any]):
```
Comment (d3rky) [inline adobe_vipm/adobe/errors.py:40]: Do you think it should be not `Any` here? It should be same type as `func` return

```
T = TypeVar('T')

def wrap_http_error(func: Callable[..., T]) -> Callable[..., T]:
  @wraps(func)
  def _wrapper(*args, **kwargs) -> T:
     ...
  return _wrapper
```
```

**Review by d3rky — CHANGES_REQUESTED** [review]

**Comment by d3rky** [comment]
@ffaraone can we think on putting all constant strings to the constants?

File: .flake8 [hunk 2]
```diff
@@ -5,4 +5,4 @@ max-line-length = 100
 application-import-names = adobe_vipm
 import-order-style = smarkets
 max-cognitive-complexity = 15
-extend-ignore = PT004
+extend-ignore = PT004, E203
```
[the remark this answers is not in the document]
Comment (ffaraone) [reply .flake8:8]: https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#slices

File: adobe_vipm/adobe/config.py [hunk 3]
```diff
@@ -26,29 +31,51 @@ def api_base_url(self) -> str:
...
+        try:
+            return self.credentials[region]
+        except KeyError:
+            raise CredentialsNotFoundError(
+                f"Credentials not found for region {region}.",
+            )
 
-    def get_reseller(self, country) -> Reseller:
-        return self.resellers.get(country)
+    def get_product(self, product_item_id: str) -> Product:
```
[the remark this answers is not in the document]
Comment (ffaraone) [reply adobe_vipm/adobe/config.py:50]: Like call it Adobe product everywhere ?

File: adobe_vipm/adobe/errors.py (hunk 1 above)
Comment (ffaraone) [reply adobe_vipm/adobe/errors.py:40]: 👍

**Review by d3rky — APPROVED** [review]

**Pull request merged**